### PR TITLE
Feature bit 37 develop icon

### PR DESCRIPTION
**Pull request by OttoVT**

### CODE REVIEW

File: src/Lykke.Job.Icon/Modules/RootstockJobModule.cs [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+                GasPriceRange = _appSettings.Nested(x => x.Job.GasPriceRange)
+            };
+
+            builder
+                .RegisterInstance(settings)
+                .SingleInstance();
+            builder.UseChainId(networkId);
+
+            builder
+                .UseAITelemetryConsumer()
```
Comment (PaterSantyago) [inline src/Lykke.Job.Icon/Modules/RootstockJobModule.cs]: You can call .UseAITelemetryConsumer() only once

File: src/Lykke.Job.Icon/Modules/RootstockJobModule.cs [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+                .RegisterType<NonceService>()
+                .As<INonceService>()
+                .SingleInstance();
+
+            builder
+                .UseAITelemetryConsumer()
+                .RegisterInstance(new HttpProvider(
+                    new HttpClient(),
```
Comment (PaterSantyago) [inline src/Lykke.Job.Icon/Modules/RootstockJobModule.cs]: It's a bad practice, HttpClientFactory is preferred

File: src/Lykke.Job.Icon/Modules/RootstockJobModule.cs [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+                .UseAITelemetryConsumer()
+                .RegisterType<KeyGenerator>()
+                .As<IKeyGenerator>()
+                .SingleInstance();
+
+            builder
+                .UseAITelemetryConsumer()
+                .RegisterType<HashCalculator>()
+                .As<IHashCalculator>()
+                .WithParameter("hashType", HashType.SHA3_256)
```
Comment (PaterSantyago) [inline src/Lykke.Job.Icon/Modules/RootstockJobModule.cs]: SHA3_256 has already been added to HashCalculator

File: src/Lykke.Service.Icon.Api.Services/BlockchainService.cs [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+            return base64;
+        }
+
+        public virtual async Task<BigInteger> EstimateGasPriceAsync()
+        {
+            var (minGasPrice, maxGasPrice) = await _gasPriceRange.GetValueAsync();
+
+            return minGasPrice + (maxGasPrice - minGasPrice) / 2;
+
+            //var estimatedGasPrice = await _apiClient.GetGasPriceAsync();
```
Comment (PaterSantyago) [inline src/Lykke.Service.Icon.Api.Services/BlockchainService.cs]: Does this commented code still necessary?

File: src/Lykke.Service.Icon.Api.Services/BlockchainService.cs [hunk 5]
```diff
@@ -0,0 +1,266 @@
...
+
+        public virtual Task<BigInteger> GetBalanceAsync(
+            string address)
+        {
+            var iconAddress = new Address(address);
+
+            return _iconService.GetBalance(iconAddress);
+        }
+
+        //?)
```
Comment (PaterSantyago) [inline src/Lykke.Service.Icon.Api.Services/BlockchainService.cs]: ?)

File: src/Lykke.Service.Icon.Api/Validators/PaginationRequestValidator.cs [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+using JetBrains.Annotations;
+using Lykke.Quintessence.Core.Utils;
+using Lykke.Quintessence.Models;
+using Microsoft.WindowsAzure.Storage.Table;
+using Newtonsoft.Json;
+
+namespace Lykke.Service.Icon.Api.Validators
+{
+    [UsedImplicitly]
+    public class PaginationRequestValidator : AbstractValidator<PaginationRequest>
```
Comment (PaterSantyago) [inline src/Lykke.Service.Icon.Api/Validators/PaginationRequestValidator.cs]: Do you can inherit from Quintessence validators?

File: src/Lykke.Job.Icon/Modules/RootstockJobModule.cs (hunk 2 above)
Comment (OttoVT) [reply src/Lykke.Job.Icon/Modules/RootstockJobModule.cs]: discussed

**Review by PaterSantyago — APPROVED** [review]

**Pull request merged**